### PR TITLE
new aws lambda extension for go

### PR DESCRIPTION
**Pull request by bardabun**
Introduced a Go integration for AWS Lambda with fresh layers.

This new file closely mirrors the pre-existing `aws-lambda-extension-node`, which has also been updated to modify commands and eliminate regional distinctions, ensuring a consistent layer version across all regions.

### CODE REVIEW

File: docs/shipping/AWS/aws-lambda-extension-go.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+id: Lambda-extension-node
```
Comment (yotamloe) [inline docs/shipping/AWS/aws-lambda-extension-go.md:2]: `node` -> `go`

File: docs/shipping/AWS/aws-lambda-extension-go.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+               `eu-central-1`, `eu-west-1`, `eu-west-2`, `eu-west-3`, `eu-north-1`,
+               `sa-east-1`,
+               `ca-central-1`.
+:::
+
+**Before you begin, you'll need**:
+  
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+* Configured [AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+* A Lambda function with a Node.js or Go application that is not yet instrumented.
```
Comment (yotamloe) [inline docs/shipping/AWS/aws-lambda-extension-go.md:31]: `node` -> `go`

File: docs/shipping/AWS/aws-lambda-extension-go.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+#### Add the OpenTelemetry collector layer to your Lambda function 
+
+This layer contains the OpenTelemetry collector that will capture data from your application.
+
+```shell
+aws lambda update-function-configuration --function-name <<YOUR-LAMBDA_FUNCTION_NAME>> --layers arn:aws:lambda:<<YOUR-AWS-REGION>>:486140753397:layer:logzio-opentelemetry-collector-<<ARCHITECHTURE>>-0_1_0:1
+```
+
+Replace `<<YOUR-LAMBDA_FUNCTION_NAME>>` with the name of your Lambda function running the Node.js application.
```
Comment (yotamloe) [inline docs/shipping/AWS/aws-lambda-extension-go.md:54]: `node` -> `go`

File: docs/shipping/AWS/aws-lambda-extension-go.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+#### Direct the OpenTelemetry collector to the configuration file
+
+
+Add `OPENTELEMETRY_COLLECTOR_CONFIG_FILE` variable to direct the OpenTelemetry collector to the configuration file:
+
+```
+aws lambda update-function-configuration --function-name <<YOUR-LAMBDA_FUNCTION_NAME>> --environment Variables={OPENTELEMETRY_COLLECTOR_CONFIG_FILE=<<PATH_TO_YOUR_COLLECTOR.YAML>>}
+```
+
+Replace `<<YOUR-LAMBDA_FUNCTION_NAME>>` with the name of your Lambda function running the Node.js application.
```
Comment (yotamloe) [inline docs/shipping/AWS/aws-lambda-extension-go.md:103]: `node` -> `go`

File: docs/shipping/AWS/aws-lambda-extension-go.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+(If `collector.yaml` is located in the root directory of your application, use the path `/var/task/collector.yaml`.)
+
+
+#### Activate tracing for your Lambda function
+
+```shell
+aws lambda update-function-configuration --function-name <<YOUR-LAMBDA_FUNCTION_NAME>> --tracing-config Mode=Active
+```
+
+Replace `<<YOUR-LAMBDA_FUNCTION_NAME>>` with the name of your Lambda function running the Node.js application.
```
Comment (yotamloe) [inline docs/shipping/AWS/aws-lambda-extension-go.md:115]: `node` -> `go`

**Review by yotamloe — CHANGES_REQUESTED** [review]
Nice job! Since as for now we dont provide an instrumentation layer, can you include an example for a go lambda function with manual instrumentation as part of this doc? Also possible to point to relevant otel go instrumentation docs

**Review by yotamloe — APPROVED** [review]
LGTM

**Comment by nico-shishkin** [comment]
Looks good, let's merge

**Pull request merged**